l10n_es_aeat_mod390_vat_prorrate/models/mod390.py

```diff
@@ -0,0 +1,139 @@
+            if item.prorrate_ids:
+                total_percent = sum(x['percent'] for x in item.prorrate_ids)
+                if total_percent == 0:
+                    raise exceptions.ValidationError(
+                        _('Sum of prorrate %  cannot be 0')
+                    )
+                elif total_percent > 100:
+                    raise exceptions.ValidationError(
+                        _('Sum of prorrate % cannot exceed 100')
+                    )
+
+    @api.multi
+    def check_prorrate_ids_amount_valid(self):
+        for item in self:
+            if item.vat_prorrate_type != 'none':
+                if item.prorrate_ids:
+                    amount = sum(x['amount'] for x in item.prorrate_ids)
+                    if amount == 0:
+                        raise exceptions.ValidationError(
+                            _('Total amount of the operations cannot be 0')
+                        )
+
+    @api.multi
+    def check_prorrate_ids_total_deduction_valid(self):
+        for item in self:
+            if item.prorrate_ids:
+                amount = sum(x['amount'] for x in item.prorrate_ids)
+                key = 'amount_allow_deduction'
+                amount_deduction = sum(x[key] for x in item.prorrate_ids)
+                if amount_deduction == 0:
+                    raise exceptions.ValidationError(
+                        _(
+                            'Total amount of the operations with the '
+                            'right to deduction cannot be 0'
+                        )
+                    )
+                elif amount_deduction > amount:
+                    raise exceptions.ValidationError(
+                        _(
+                            'Total amount of the operations with the '
+                            'right to deduct cannot exceed the amount of '
+                            'the operations'
+                        )
+                    )
+
+    @api.multi
+    def button_confirm(self):
+        for item in self:
+            if item.prorrate_ids:
+                if item.vat_prorrate_type != 'none':
+                    item.vat_prorrate_type = 'none'
+
+                item.check_prorrate_ids_percent_valid()
+                item.check_prorrate_ids_total_deduction_valid()
+
+            item.check_prorrate_ids_amount_valid()
+
+        return super(L10nEsAeatMod390Report, self).button_confirm()
+
+    @api.multi
+    @api.onchange('vat_prorrate_type')
+    def _onchange_vat_prorrate_types(self):
+        for item in self:
+            if item.vat_prorrate_type == 'none':
+                item.vat_prorrate_percent = 0
+
+    @api.constrains('vat_prorrate_percent')
+    def check_vat_prorrate_percent(self):
+        if self.vat_prorrate_type != 'none':
+            if not (0 < self.vat_prorrate_percent <= 100):
+                raise exceptions.ValidationError(
+                    _('VAT prorrate percent must be between 0.01 and 100')
+                )
```

Comment:
```suggestion
        for item in self:
            if item.vat_prorrate_type != 'none':
                if not (0 < item.vat_prorrate_percent <= 100):
                    raise exceptions.ValidationError(
                        _('VAT prorrate percent must be between 0.01 and 100')
                    )
```
To avoid error if write or create method is invoked with a multiple recordset